internal/resolution/variablesources/required_package.go

```diff
@@ -1,104 +1,53 @@
 package variablesources
 
 import (
-	"context"
 	"fmt"
 	"sort"
 
 	mmsemver "github.com/Masterminds/semver/v3"
-	"github.com/operator-framework/deppy/pkg/deppy"
-	"github.com/operator-framework/deppy/pkg/deppy/input"
 
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 	catalogfilter "github.com/operator-framework/operator-controller/internal/catalogmetadata/filter"
 	catalogsort "github.com/operator-framework/operator-controller/internal/catalogmetadata/sort"
 	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
 )
 
-var _ input.VariableSource = &RequiredPackageVariableSource{}
-
-type RequiredPackageVariableSourceOption func(*RequiredPackageVariableSource) error
-
-func InVersionRange(versionRange string) RequiredPackageVariableSourceOption {
-	return func(r *RequiredPackageVariableSource) error {
-		if versionRange != "" {
-			vr, err := mmsemver.NewConstraint(versionRange)
-			if err == nil {
-				r.versionRange = versionRange
-				r.predicates = append(r.predicates, catalogfilter.InMastermindsSemverRange(vr))
-				return nil
-			}
-
-			return fmt.Errorf("invalid version range '%s': %w", versionRange, err)
-		}
-		return nil
-	}
-}
-
-func InChannel(channelName string) RequiredPackageVariableSourceOption {
-	return func(r *RequiredPackageVariableSource) error {
-		if channelName != "" {
-			r.channelName = channelName
-			r.predicates = append(r.predicates, catalogfilter.InChannel(channelName))
-		}
-		return nil
+// RequiredPackageVariable returns a variable which represent
+// explicit requirement for a package from an user.
+// This is when an user explicitly asks "install this" via Operator API.
+func RequiredPackageVariable(allBundles []*catalogmetadata.Bundle, packageName, channelName, versionRange string) (*olmvariables.RequiredPackageVariable, error) {
```

Comment:
This re-uses a bits from `RequiredPackageVariableSource`. What was removed:

* `NewRequiredPackageVariableSource` & `RequiredPackageVariableSourceOption`. We just pass things as arguments into function without options abstraction
* Separate `notFoundError` function for formatting an error message